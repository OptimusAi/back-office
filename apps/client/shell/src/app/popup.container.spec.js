import 'regenerator-runtime/runtime';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import 'jest-canvas-mock';
import configureStore from 'redux-mock-store';
import Popup from './popup.container';
import SpecFriendly from '../spec-friendly';

const mockStore = configureStore([]);
describe('Pop up Container', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      app: {
        management: {
          zones: { error: { message: 'Error', type: 'error', open: true } },
          zoneCategories: {
            error: { message: 'Error', type: 'error', open: true },
          },
          bylaws: {
            error: { message: 'Error', type: 'error', open: true },
          },
          enforcement: {
            infractionsByZone: {
              error: { message: 'Error', type: 'error', open: true },
            },
          },
        },
        error: { message: 'Error', type: 'error', open: true },
      },
    });
    component = render(
      <SpecFriendly>
        <Provider store={store}>
          <Popup />
        </Provider>
      </SpecFriendly>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component).toMatchSnapshot();
  });
});
