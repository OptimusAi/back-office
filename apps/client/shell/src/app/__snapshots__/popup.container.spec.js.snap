// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pop up Container should render with given state from Redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
      >
        <div
          class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation6"
          direction="up"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message"
          >
            Error
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-filledError MuiPaper-elevation6"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          Error
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
